python/demos/demo_periodic_gep.py

```diff
@@ -198,82 +205,143 @@ def solve_GEP_shiftinvert(A: PETSc.Mat, B: PETSc.Mat,
+        pbc_hor_mt = meshtags(mesh, fdim, facets[arg_sort], np.full(len(facets), pbc_hor_slave_tag, dtype=np.int32))
+        
+    # Periodic boundary condition: {y=0} -> {y=1}
+    if boundary_condition[1]=="periodic":
+        pbc_vert_slave_tag = 3
+        pbc_vert_is_slave = lambda x: np.isclose(x[1],1)
+        def pbc_vert_slave_to_master_map(x):
+            out_x = np.zeros(x.shape)
+            out_x[0] = x[0]
+            out_x[1] = x[1] - 1
+            out_x[2] = x[2]
+            return out_x
+        facets = locate_entities_boundary(mesh, fdim, pbc_vert_is_slave)
+        arg_sort = np.argsort(facets)
+        pbc_vert_mt = meshtags(mesh, fdim, facets[arg_sort], np.full(len(facets), pbc_vert_slave_tag, dtype=np.int32))
+        
+    # Create MultiPointConstraint object
+    mpc = MultiPointConstraint(V)
+    if boundary_condition[0]=="periodic":
+        mpc.create_periodic_constraint_topological(V, pbc_hor_mt, 
+                                                   pbc_hor_slave_tag,
+                                                   pbc_hor_slave_to_master_map, 
+                                                   bcs)
+    if boundary_condition[1]=="periodic":
+        if boundary_condition[0]=="periodic":
+            # Redefine slave_to_master map to exclude dofs that are slave or 
+            # master of the previous periodic boundary condition
+            tmp = pbc_vert_slave_to_master_map
+            def pbc_vert_slave_to_master_map(x):
+                out_x = tmp(x)
+                idx = pbc_hor_is_slave(x) + pbc_hor_is_master(x)
+                out_x[0][idx] = np.nan
+                return out_x
+        mpc.create_periodic_constraint_topological(V, pbc_vert_mt, 
+                                                   pbc_vert_slave_tag,
+                                                   pbc_vert_slave_to_master_map, 
+                                                   bcs)
+    mpc.finalize()
+    # Define variational problem
+    u = TrialFunction(V)
+    v = TestFunction(V)
+    a = inner(grad(u), grad(v)) * dx
+    b = inner(u, v) * dx
+    mass_form = fem.form(a)
+    stiffness_form = fem.form(b)
+    # Diagonal values for slave and Dirichlet DoF
+    # The generalized eigenvalue problem will have spurious eigenvalues at
+    # lambda_spurious = diagval_A/diagval_B. Here we choose lambda_spurious=1e4,
+    # which is far from the region of interest.
+    diagval_A = 1e2
+    diagval_B = 1e-2
+    A = assemble_matrix(mass_form, mpc, bcs=bcs, diagval=diagval_A)
+    B = assemble_matrix(stiffness_form, mpc, bcs=bcs, diagval=diagval_B)
+    EPS = solve_GEP_shiftinvert(A, B, problem_type=SLEPc.EPS.ProblemType.GHEP,
+                                solver=SLEPc.EPS.Type.KRYLOVSCHUR,
+                                nev=Nev, tol=1e-7, max_it=10,
+                                target=1.5, shift=1.5,comm=comm)
+    (eigval, eigvec_r, eigvec_i) = EPS_get_spectrum(EPS, mpc)
+    # update slave DoF
```

Comment:
Add blank line between above lines with only comments

Reply:
On the current fmonteghetti:master, flake8 does not highlight any issues.